core/chains/evm/client/client.go

```diff
@@ -366,3 +367,7 @@ func (client *client) SuggestGasTipCap(ctx context.Context) (tipCap *big.Int, er
 func (client *client) IsL2() bool {
 	return client.pool.ChainType().IsL2()
 }
+
+func (client *client) LatestFinalizedBlock(_ context.Context) (*evmtypes.Head, error) {
+	panic("not implemented. client was deprecated. New methods are added only to satisfy type constraints while we are migrating to new alternatives")
```

Comment:
I don't think the entire app should crash if someone uses this method by mistake. It's better if we return nil & an error message so any user can handle this gracefully.